network-api/networkapi/wagtailpages/wagtail_hooks.py

```diff
@@ -94,3 +94,7 @@ def before_delete_page(request, page):
             pass
         else:
             uploader.destroy(page.cloudinary_image.public_id, invalidate=True)
+
+    if hasattr(page, 'votes'):
```

Comment:
Let's make this a little stronger because we don't want to delete `votes` from "Any and all page instances, no matter how deeply subclasses", we only want to delete `votes` from ProductPage instances, so let's at the very least check `isinstance(page, ProductPage) and page.votes`?